File: Javascript/Basics/topic c - arrays and control flow/fibonacciSequenceGenerator.js

// a function that generate a specified (n) number of values in a Fibonacci Sequence
function fibonacciGenerator(n) {
    if (n === 1) {
        let fbSeries = [0];
        return fbSeries;
        
    } else if (n === 2) {
        let fbSeries = [0, 1];
        return fbSeries;
        
    } else {
        let fbSeries = [0, 1]; 
        /*
            - getting the last 2 values in the array
            - adding the values
            - pushing their sum to the end of the array until the value of 'n' is equal to the length of the array
        */
        let lastValue = (fbSeries[fbSeries.length - 1] + fbSeries[fbSeries.length - 2]);
        fbSeries.push(lastValue);

        // looping through the array to add a specified number of values until the value of 'n' is equal to the length of the array
        for (let x = 2; x < n; x++) {
            let lastValue = (fbSeries[fbSeries.length - 1] + fbSeries[fbSeries.length - 2]);
            fbSeries.push(lastValue);
        }
        return fbSeries;
    }
}

fibonacciGenerator(6);
 

File: Python/projects/fibonacciSequenceGenerator.py
# Fibonacci Sequence Generator Function
def fibonacciGenerator(n):
    if n == 1:
        output = [0]
        print(output)
    elif n == 2:
        output = [0, 1]
        print(output)

    else:
        output = [0, 1]
        listLen = len(output)

        # looping through the list until the (n) number of fibonnacci values in the sequence is generated
        while listLen < n:
            lastValue = output[listLen - 1] + output[listLen - 2]  # getting the sum of the last 2 values in the current list postion
            output.append(lastValue)     # adding the resulting sum to the end of the list
            print(output)
            listLen += 1    # increamenting the length of the list until it's equal to (n)

fibonacciGenerator(10)